Tiloaria/assignment4/Islands.cpp

```diff
@@ -0,0 +1,97 @@
+#include <cstdio>
+#include <cstdlib>
+#include <iostream>
+#include <set>
+#include <vector>
+#include <string>
+#include <assert.h>
+#include <algorithm>
+
+using namespace std;
+
+/*Set value false for all tiles which belong current island.
+Used DFS algorithm.
+In this function map is not empty*/
+void clear_this_island(vector<vector<bool> > &helper_map, int w, int h) {
+    int width = helper_map[0].size();
```

Comment:
The comment from the `count_islands` function also applies here :)

Generally, you want to guard your implementation around the 'public' interfaces, i.e. if we assume that `clear_this_island` will only be called from `count_islands` and never as a stand-alone function, then we're ok assuming the map has the same number of rows throughout if we check the dimensions properly in `count_islands` :)